### PR TITLE
Questionnaire

### PR DESCRIPTION
**Pull request by garamond**
localhost_3010_umfrage_2018 1

## Todo
- [x] Network error handling
- [x] Externalize strings
- [x] Input validation
- UI
  - [x] Placeholder in `ArticleQuestion`
  - [x] Slider styling for FF & IE11
- Meta page
  - [x] redirect to `/verlag` after submission
- [x] only show before / after on `/verlag` when page loaded
- [x] fill in gallery content
- [x] look into optimistic updates for checkboxes and radio
- [ ] check chart colors

@garamond Suggestions re results:

- [ ] Make bar of winning candidate for president larger (by showing percentage)
- [ ] Show percentages instead of absolute numbers

### CODE REVIEW

File: components/Questionnaire/TextInput/TextInput.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        />
+      </div>
+    )
+  }
+
+  render () {
+    const { text, placeholder, onChange } = this.props
+    return (
+      <div {...styles.form}>
+        <Textarea
```
Comment (tpreusse) [inline components/Questionnaire/TextInput/TextInput.js]: Use a styleguide field with a custom `renderInput`:
```
import { styles as fieldSetStyles } from 'components/FieldSet'

<Field {...}
  renderInput={({ ref, ...inputProps }) => (
    <AutosizeInput {...fieldSetStyles.autoSize}
      {...inputProps}
      inputRef={ref} />
    )}
 />
```

This way you'll get a bottom border and unified input look.

File: components/Questionnaire/Page.js [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+          // handle already submitted
+          const { questionnaire: { userHasSubmitted, questions } } = data
+          const { error, submitting, updating } = this.state
+          if (!submitting && userHasSubmitted) {
+            return (
+              <>
+                <Headline>{t('questionnaire/title')}</Headline>
+                <div {...styles.thankyou}>
+                  <P>
+                    {t('questionnaire/thankyou')}
```
Comment (tpreusse) [inline components/Questionnaire/Page.js]:
```suggestion
                    {t.elements('questionnaire/thankyou', {
                      metaLink: <Link key='meta' route='/verlag' passHref>
                        <A>{t('questionnaire/thankyou/metaText')}</A>
                      </Link>
                    })}
```

File: components/Questionnaire/Page.js [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+              </>
+            )
+          }
+
+          // handle questions
+          const questionCount = questions.filter(Boolean).length
+          const userAnswerCount = questions.map(q => q.userAnswer).filter(Boolean).length
+          return (
+            <div>
+              <Headline>Umfrage</Headline>
```
Comment (tpreusse) [inline components/Questionnaire/Page.js]: move to translations :)

File: components/Questionnaire/Page.js [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+              }
+            }
+          }
+        }
+        return mutate({
+          variables: {
+            questionId,
+            payload
+          },
+          ...(answerId ? optimistic : {})
```
Comment (tpreusse) [inline components/Questionnaire/Page.js]: this leads to initial race conditions. we should do client side uuid generation, let's discuss with @patte

File: components/Questionnaire/Page.js (hunk 2 above)
Comment (garamond) [reply components/Questionnaire/Page.js]: nice

**Pull request merged**